tagstudio/src/qt/widgets/video_player.py

```diff
@@ -0,0 +1,377 @@
+        self.animation.setEndValue(0)
+        self.animation.setDuration(500)
+        self.animation.start()
+        self.play_pause.hide()
+        self.mute_button.hide()
+        # self.fullscreen_button.hide()
+        return super().releaseMouse()
+
+    def resetControlsToDefault(self) -> None:
+        # Resets the video controls to their default state.
+        self.play_pause.load("./tagstudio/resources/pause.svg")
+        self.mute_button.load("./tagstudio/resources/volume_muted.svg")
+
+    def pauseToggle(self) -> None:
+        if self.player.isPlaying():
+            self.player.pause()
+            self.play_pause.load("./tagstudio/resources/play.svg")
+        else:
+            self.player.play()
+            self.play_pause.load("./tagstudio/resources/pause.svg")
+
+    def muteToggle(self) -> None:
+        if self.player.audioOutput().isMuted():
+            self.player.audioOutput().setMuted(False)
+            self.mute_button.load("./tagstudio/resources/volume_unmuted.svg")
+        else:
+            self.player.audioOutput().setMuted(True)
+            self.mute_button.load("./tagstudio/resources/volume_muted.svg")
+
+    def play(self, filepath: str, resolution: QSize) -> None:
+        # Sets the filepath and sends the current player position to the very end, so that the new video can be played.
+        # self.player.audioOutput().setMuted(True)
+        logging.info(f"Playing {filepath}")
+        self.resolution = resolution
+        self.filepath = filepath
+        if self.player.isPlaying():
+            self.player.setPosition(self.player.duration())
+            self.player.play()
+        else:
+            self.checkMediaStatus(QMediaPlayer.MediaStatus.EndOfMedia)
+        # logging.info(f"Successfully stopped.")
+
+    def stop(self) -> None:
+        self.filepath = None
+        self.player.stop()
+
+    def resizeVideo(self, new_size: QSize) -> None:
+        # Resizes the video preview to the new size.
+        self.video_preview.setSize(new_size)
+        self.video_tint.setRect(
+            0, 0, self.video_preview.size().width(), self.video_preview.size().height()
+        )
+
+        rcontent = self.contentsRect()
+        self.centerOn(self.video_preview)
+        self.roundCorners()
+        self.setSceneRect(0, 0, rcontent.width(), rcontent.height())
+        self.keepControlsInPlace()
+
+    def roundCorners(self) -> None:
+        width: int = int(max(self.contentsRect().size().width(), 0))
+        height: int = int(max(self.contentsRect().size().height(), 0))
+        mask = Image.new(
+            "RGBA",
+            (
+                width,
+                height,
+            ),
+            (0, 0, 0, 255),
+        )
+        draw = ImageDraw.Draw(mask)
+        draw.rounded_rectangle(
+            (0, 0) + (width, height),
+            radius=12,
+            fill=(0, 0, 0, 0),
+        )
+        mask = mask.getchannel("A").toqpixmap()
```

Comment:
`mask` has originally type `PIL.Image`, but here it's changed to `QPixmap`, so it would be really better to use different variable for one of these, so the types wont clash